examples/lp503x/lp503x_main.c

```diff
@@ -0,0 +1,716 @@
+
+  numleds = atoi(parg);
+  if ((numleds > 12) || (numleds < 1))
+    {
+      printf("Error: number of leds should be between 1 and 12\n");
+      return -EINVAL;
+    }
+
+  curcolour = 0;
+
+  for (i = 0; i < 5; i++)
+    {
+      for (j = 0; j < numleds; j++)
+        {
+          lp503x_ioctl_args->lednum = j;
+          lp503x_ioctl_args->param = pattern_brightness[i];
+          ret = ioctl(fd, PWMIOC_SET_RGB_BRIGHTNESS, lp503x_ioctl_args);
+          curcolour++;
+          if (curcolour >= numleds)
+            {
+              curcolour = 0;
+            }
+
+          lp503x_ioctl_args->param = colour_lookup[curcolour];
+          ret = ioctl(fd, PWMIOC_SET_RGB_COLOUR, lp503x_ioctl_args);
+        }
+
+      usleep(500000);
+      curcolour++;
+    }
+
+  lp503x_ioctl_args->param = 0;
+  for (k = 0; k < numleds; k++)
+    {
+      lp503x_ioctl_args->lednum = k;
+      ret = ioctl(fd, PWMIOC_SET_RGB_COLOUR, lp503x_ioctl_args);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name:lp503x_cmd_set_bank_brightness
+ *
+ *   sets the bank brightness to be used
+ ****************************************************************************/
+
+static int lp503x_cmd_set_bank_brightness(char *parg)
+{
+  int bank_brightness;
+  if (parg != NULL || *parg != '\0')
+    {
+      bank_brightness = atoi(parg);
+
+      if ((bank_brightness > MAX_BRIGHTNESS) || (bank_brightness < 0))
+        {
+          printf("bank brightness range is 0..255\n");
+          bank_brightness = 0;
+          return -EINVAL;
+        }
+      else
+        {
+          lp503x_ioctl_args->param = bank_brightness;
+          current_led_brightness = bank_brightness;
+          return ioctl(fd, PWMIOC_SET_BANK_BRIGHTNESS, lp503x_ioctl_args);
+        }
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name:lp503x_setcolour
+ *
+ *   sets the current RGB colour to be used
+ ****************************************************************************/
+
+static int lp503x_cmd_setcolour(char *parg)
+{
+  int colour;
+  if (parg == NULL || *parg != '\0')
+    {
+      printf("current colour is: 0x%06x\n", current_colour);
+    }
+  else
+    {
+      colour = (int)strtol(parg, NULL, 16);
+
+      if (colour > MAX_RGB_COLOUR)
+        {
+          printf("colour range is 0...0xFFFFFF\n");
+          current_colour = 0;
+          return -EINVAL;
+        }
+      else
+        {
+          current_colour = colour;
+          printf("colour set to: 0x%06x\n", current_colour);
+        }
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: lp503x_cmd_help
+ *
+ *   lp503x_cmd_help() displays the application's help information on
+ *   supported commands and command syntax.
+ ****************************************************************************/
+
+static int lp503x_cmd_help(char *parg)
```

Comment:
```suggestion
static int lp503x_cmd_help(FAR char *parg)
```

Reply:
is the general rule then:

- things that could be used for any arch, 8/16/32/64 bit, we need the FAR
- drivers etc that are specific to a 32/64 bit arch don't need FAR?

Only ever developed for 32bit processors so this is a new (but old!) one for me.